From 76b8ec3df5a9529469a684f2022cd80e05089488 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@freescale.com>
Date: Wed, 20 May 2015 16:25:00 +0800
Subject: [PATCH] MLK-10969-4 video: mxsfb: refine lcdif driver to support
 mipi dsi

Refine some code of lcdif driver to support mipi dsi
correctly.

Signed-off-by: Fancy Fang <chen.fang@freescale.com>
---
 drivers/video/mxsfb.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/video/mxsfb.c b/drivers/video/mxsfb.c
index afb4007..ec8ebe7 100644
--- a/drivers/video/mxsfb.c
+++ b/drivers/video/mxsfb.c
@@ -509,6 +509,7 @@ static void mxsfb_enable_controller(struct fb_info *fb_info)
 				"dispdrv:%s\n", host->dispdrv->drv->name);
 			return;
 		}
+		host->sync = fb_info->var.sync;
 	}
 
 	if (host->reg_lcd) {
@@ -868,9 +869,10 @@ static int mxsfb_blank(int blank, struct fb_info *fb_info)
 		break;
 
 	case FB_BLANK_UNBLANK:
-		if (!host->enabled)
+		if (!host->enabled) {
+			mxsfb_set_par(&host->fb_info);
 			mxsfb_enable_controller(fb_info);
-		mxsfb_set_par(&host->fb_info);
+		}
 		break;
 	}
 	return 0;
@@ -1223,6 +1225,7 @@ static void mxsfb_dispdrv_init(struct platform_device *pdev,
 	struct device *dev = &pdev->dev;
 	char disp_dev[32];
 
+	memset(&setting, 0x0, sizeof(setting));
 	setting.fbi = fbi;
 	memcpy(disp_dev, host->disp_dev, strlen(host->disp_dev));
 	disp_dev[strlen(host->disp_dev)] = '\0';
-- 
1.8.0

